### PR TITLE
config: add REST API Protocol to LanguageTool

### PR DESCRIPTION
**Pull request by hcvcastro**
Change-Id: I0e05ed89d02a9b467e1707db1638fc9ee554586a
Signed-off-by: Henry Castro <hcastro@collabora.com>

### CODE REVIEW

File: wsd/COOLWSD.cpp [hunk 1]
```diff
@@ -2418,6 +2419,8 @@ void COOLWSD::innerInitialize(Application& self)
     setenv("LANGUAGETOOL_APIKEY", apiKey.c_str(), 1);
     bool sslVerification = getConfigValue<bool>(conf, "languagetool.ssl_verification", "");
     setenv("LANGUAGETOOL_SSL_VERIFICATION", sslVerification ? "true" : "false", 1);
+    const std::string restProtocol = getConfigValue<std::string>(conf, "languagetool.rest_protocol", "");
+    setenv("LANGUAGETOOL_RESTPROTOCOL", restProtocol.c_str(), 1);
```
Comment (Ashod) [inline wsd/COOLWSD.cpp:2423]: We shouldn't need envars (same for previous ones). The kit does get the contents of the coolwsd.xml file. See ConfigUtil and how it's used in Kit.cpp and ForKit.cpp by searching for `config::`. It's as easy as doing `config::getString("languagetool.rest_protocol", "");`

**Review by Ashod — COMMENTED** [review]
Thanks @hcvcastro. One note.

File: wsd/COOLWSD.cpp (hunk 1 above)
Comment (Ashod) [reply wsd/COOLWSD.cpp:2423]: I can see now that this envar is used in Core. Still, if we must use the envar, best to set it in ForKit and not here in WSD. I think that's cleaner. Same for other LANGUAGETOOL envars.

Comment (hcvcastro) [reply wsd/COOLWSD.cpp:2423]: I agree, but it is not my requirements to touch those LanguageTools settings. Maybe someone can do the followup clean patch :)

Comment (Ashod) [reply wsd/COOLWSD.cpp:2423]: Surely we can move these `setenv` calls to ForKit. That's all I'm suggesting. But we can do that in a follow up.

**Review by gokaysatir — APPROVED** [review]
Thanks for the improvements :)

**Pull request merged**